drivers/flash/spi_flash_w25qxxdv.c

```diff
@@ -146,24 +146,53 @@ static int spi_flash_wb_read(struct device *dev, off_t offset, void *data,
 	return ret;
 }
 
-static int spi_flash_wb_write(struct device *dev, off_t offset,
-			      const void *data, size_t len)
+static int spi_flash_wb_write_protection_set_with_lock(struct device *dev,
+						       bool enable, bool lock)
 {
 	struct spi_flash_data *const driver_data = dev->driver_data;
-	u8_t reg;
+	u8_t reg = 0U;
 	int ret;
 
-	if (offset < 0) {
-		return -ENOTSUP;
+	if (lock) {
```

Comment:
Would it no be a lot easier to replace the k_sem with a K_mutext which is re-entrant than we don't have to worry if we lock twice we just have to unlock the same amount